src/query/interpreter.cpp

```diff
@@ -286,39 +305,29 @@ class ReplQueryHandler final : public query::ReplicationQueryHandler {
   void RegisterReplica(const std::string &name, const std::string &socket_address,
                        const ReplicationQuery::SyncMode sync_mode,
                        const std::chrono::seconds replica_check_frequency) override {
-    if (db_->GetReplicationRole() == storage::replication::ReplicationRole::REPLICA) {
+    if (handler_.IsReplica()) {
       // replica can't register another replica
       throw QueryRuntimeException("Replica can't register another replica!");
     }
 
-    if (name == storage::replication::kReservedReplicationRoleName) {
+    if (name == memgraph::replication::kReservedReplicationRoleName) {
       throw QueryRuntimeException("This replica name is reserved and can not be used as replica name!");
     }
 
-    storage::replication::ReplicationMode repl_mode;
-    switch (sync_mode) {
-      case ReplicationQuery::SyncMode::ASYNC: {
-        repl_mode = storage::replication::ReplicationMode::ASYNC;
-        break;
-      }
-      case ReplicationQuery::SyncMode::SYNC: {
-        repl_mode = storage::replication::ReplicationMode::SYNC;
-        break;
-      }
-    }
+    auto repl_mode = convertToReplicationMode(sync_mode);
```

Comment:
IIFE would be better IMO since the lambda is used just once.

Reply:
One change I made is to make it `inline`. 

Even though it is only used once, I want to pull out these replication -> query, and query -> replication conversions into their own functions. In order to make the functions that use them easy to follow.

i.e. don't want the lambda to distract from the rest of the code.